### DISCUSSION TITLE
No Predictions with a Bottom Up-ID model

### DISCUSSION THREAD

**Discussion by cece3014** [talmolab/sleap#2477] [Help!]
Hi SLEAP team!
I have been working in SLEAP the last few months and am truly amazed by the results of the bottom-up model I have been able to train (thank you to @gitttt-1234 for so much guidance). For my particular dataset that is divided into age classes, I would love to compare how the bottom up ID model performs in relation to a pose-estimation only model. When i ran a training with IDs attached to the instances, I got 0 predictions. Perhaps there are different model hyperparameters that would be optimal for an ID model that are different than a bottom-up model?  

I would appreciate any insight you have, thank you again.

**Example image of dataset**
(all of these predictions were made using the bottom-up model, not an ID mdoel)
<img width="3806" height="2014" alt="Screenshot 2025-11-04 103903" src="https://github.com/user-attachments/assets/04afc76a-acac-418c-899c-114911996001" />
<img width="3827" height="1999" alt="Screenshot 2025-11-04 103207" src="https://github.com/user-attachments/assets/fe3807d4-3d34-4071-a4f3-6eee465f904c" />
<img width="1902" height="1010" alt="250130_M3SP_S5_U3_DJI_0045_0046_Trim1_f813" src="https://github.com/user-attachments/assets/40d404d3-2226-42bb-8102-6d401567abf9" />

**Model Hyperparameters for bottom-up model**
<img width="3839" height="2159" alt="Screenshot 2025-10-31 130518" src="https://github.com/user-attachments/assets/db4c656d-62f8-4da5-a927-3ab515605413" />

**Model Hyperparameters for bottom-up ID model**
<img width="3839" height="2052" alt="mult-animal-bottom-up-IDmodel-hyperparameters" src="https://github.com/user-attachments/assets/65d6f831-0d9a-4636-a1be-645d50c4a4a8" />

**Comment by gitttt-1234** [comment]
Hi @cece3014 !

Could you let us know how the training curves looked like? you could share the `training_log.csv` [here](https://docs.google.com/forms/d/e/1FAIpQLSfWBwcuZH8lxGyFabMAODLjSUE4YrSCYbrrnYvXCpgBqZpcJA/viewform) so that we could view the loss curves. Based on your config, I’d suggest trying the following changes: Enable rotation augmentations — without them, the model can overfit to the exact poses seen during training. In the model config, check the up-interpolate option and in the heads section, set the second sigma to 15.0 and the second loss weight to 2.0. Give these a try and let us know if the predictions improve!

Thanks,

Divya

**Reply by cece3014** [reply]
Hi @gitttt-1234 

Good morning - Absolutely! You mean the training_log for the bottom_up_IDmodel correct? 

Is it only possible to create one ID model? I wanted to attempt a new modl using the parameters suggested but I can only select the previous model that produced 0 results. 

Thank you.
<img width="1906" height="1042" alt="bottom_up_IDmodel_example" src="https://github.com/user-attachments/assets/fac4acf8-0f49-49f0-8ac9-b161a5d6ca5d" />

**Reply by gitttt-1234** [reply]
Hi @cece3014 !

You don't have to choose anything in the drop down, but just update the values that should work!

Thanks,

Divya

**Reply by cece3014** [reply]
Hi @gitttt-1234,

I hope your holiday season was eventful! I adjusted the model configurations as suggested for training but unfortunately had 0 predictions. Are there other options to explore?

<img width="3838" height="2044" alt="Screenshot 2025-12-03 104050" src="https://github.com/user-attachments/assets/86fdd1c0-b9f4-481a-9fe7-5f2afb93c4f8" />

**Reply by gitttt-1234** [reply]
Hi @cece3014 !

Apologies for the delay!

How many epochs did the model train for and how did the training curves look like? Could you share the training_log.csv [here](https://docs.google.com/forms/d/e/1FAIpQLSfWBwcuZH8lxGyFabMAODLjSUE4YrSCYbrrnYvXCpgBqZpcJA/viewform)? If the model has only trained for very few epochs, then try adjusting the early stopping (set delta as 1e-8 / increase the patience) as training with multiple heads can be little unstable and the default parameters might not work! In your previous model, you set input scaling: 0.25 and output stride: 4. Give it a try with these same parameters and let us know if that works!

Thanks,

Divya

**Reply by cece3014** [reply]
Hi @gitttt-1234 ,

Not a problem at all, thank you for your suggestions! I uploaded the previous bottom-up ID model's training_log csv to the link provided. In the meantime I ran another bottom-up ID with theses adjustments, but still no predictions. Would you suggest to continue to adjust early stopping?

Thank you so much.
